src/core.js

```diff
@@ -1395,7 +1395,7 @@ $.extend( $.validator, {
 			if ( notSupported ) {
 				throw new Error( errorMessage );
 			}
-			return this.optional( element ) || ( value % param === 0 );
+			return this.optional( element ) || ( (value / param) % 1 === 0 );
```

Comment:
This will fail for `0.07` because:
```js
(0.07 / 0.01) % 1
//> 8.881784197001252e-16
```

To fix that, we can do:
```diff
@@ -1386,18 +1386,44 @@ $.extend( $.validator, {
     step: function( value, element, param ) {
       var type = $( element ).attr( "type" ),
         errorMessage = "Step attribute on input type " + type + " is not supported.",
         supportedTypes = [ "text", "number", "range" ],
         re = new RegExp( "\\b" + type + "\\b" ),
-        notSupported = type && !re.test( supportedTypes.join() );
+        notSupported = type && !re.test( supportedTypes.join() ),
+        decimalPlaces = function( num ) {
+          var match = ( '' + num ).match( /(?:\.(\d+))?$/ );
+          if (!match) {
+            return 0;
+          }
+
+          // Number of digits right of decimal point.
+          return match[1] ? match[1].length : 0;
+        },
+        toInt = function( num ) {
+          return Math.round( num * Math.pow( 10, decimals ) );
+        },
+        valid = true,
+        decimals;
 
       // Works only for text, number and range input types
       // TODO find a way to support input types date, datetime, datetime-local, month, time and week
       if ( notSupported ) {
         throw new Error( errorMessage );
       }
-      return this.optional( element ) || ( value % param === 0 );
+
+      decimals = decimalPlaces( param );
+
+      // Value can't have too many decimals
+      if ( decimalPlaces( value ) > decimals ) {
+        valid = false;
+      }
+
+      if ( toInt( value ) % toInt( param ) != 0 ) {
+        valid = false;
+      }
+
+      return this.optional(element) || valid;
     },
```